src/box/applier.cc

```diff
@@ -2258,6 +2267,7 @@ applier_f(va_list ap)
 			applier_disconnect(applier, APPLIER_LOADING);
 			goto reconnect;
 		} catch (FiberIsCancelled *e) {
+			ERROR_INJECT_YIELD(ERRINJ_APPLIER_DESTROY_DELAY);
```

Comment:
The problem with both those error injections is that they are added to impossible places. Meaning that without those injections there couldn't be any yields at those lines. Basically, you introduce a bug with the new yields and then fix it, alongside apparently fixing the original bug too.

Instead, error injections should reflect what can happen in reality. Since you needed only a delay for applier fiber join, you could add the injection right before or after that fiber join. Or add a usleep() injection inside the applier thread where needed. These would actually simulate something that can happen in reality.

Reply:
Finally manged to write a 100% reproducer with good error injections. Thanks for the hint)

Reply:
Thank you!